crates/mun_abi/src/struct_info.rs

```diff
@@ -88,22 +92,81 @@ impl From<StructMemoryKind> for u64 {
     }
 }
 
+impl<'a> PartialEq for StructInfo<'a> {
+    fn eq(&self, other: &Self) -> bool {
```

Comment:
An interesting question here is: do we want to validate that all fields are equal, or do we want a performant equality operator? Just checking `GUID`s will guarantee that two structs are not the same; and is much faster!

Reply:
But they don't (yet) include the offsets right?

Reply:
👍 Ill change it to GUID compare then.